modules/common/src/main/scala/emil/Disposition.scala

```diff
@@ -0,0 +1,22 @@
+package emil
+
+sealed trait Disposition {def name: String}
+
+/**
+  * Enumeration of disposition types for a Content-Disposition part header.
+  */
+object Disposition {
+  case object Inline extends Disposition {val name = "inline"}
+  case object Attachment extends Disposition {val name = "attachment"}
+
+  val values = Seq(Inline, Attachment)
+
+  /**
+    * Fail save disposition parser.
+    *
+    * @param name name of the disposition type to parse
+    * @return found disposition type or None
+    */
+  def withName(name: String): Option[Disposition] = values.find(v => v.name.equalsIgnoreCase(name))
```

Comment:
I usually use (also in emil) the name `fromString` for this method. This could return an `Either[String, Disposition]` so a caller could get a better error message (for debugging it is sometimes useful to know the value that failed).  For an example you can look into `SSLType`. (only for the name and return type, the impl is great). Callers can convert easily to `Option` via `.toOption`. Again, not a merge stopper!

Reply:
Yes, I can follow your objection. I just had a look at the concept with "Either" and that feels much better. 

When it comes to naming factory methods, I'm open to what the conventions dictate. Consistency is important there.